### PR TITLE
Update plots.py

### PR DESCRIPTION
**Pull request by HMEUW**
Added new function section_plot.

### CODE REVIEW

**Comment by OnnoEbbens** [comment]
Thanks for the PR! Before we merge could you add these two things?
1. An example of this function to an existing noteook ('01_groundwater_observations.ipynb')
2. A test function in the module `test_008_visualisation.py`

Let me know if you have any questions or you want so help doing this

File: hydropandas/extensions/plots.py [hunk 1]
```diff
@@ -307,6 +307,247 @@ def interactive_map(
...
+            by_label = dict(zip(labels, handles))
+            ax.legend(by_label.values(), by_label.keys())
+
+        fig.suptitle(self._obj.name)
+
+        if fn_save is not None:
+            try:
+                fig.savefig(fn_save, dpi=250, bbox_inches='tight',)
+            except:
+                logger.error(f"Save of figure {o.name} failed ({fn_save}).")
```
Comment (dbrakenhoff) [inline hydropandas/extensions/plots.py:547]: variable `o` here does not exist, should be `self._obj`?

File: hydropandas/extensions/plots.py [hunk 2]
```diff
@@ -307,6 +307,247 @@ def interactive_map(
...
+                                   )
+
+            # PART 2: fancy section plot with lines along tube
+
+            # highlight filter on section plot
+            ax_section.plot([plot_x[counter]]*2,
+                            [self._obj.loc[name, "screen_top"], self._obj.loc[name, "screen_bottom"]],
+                            color='k',
+                            lw=3,
+                            ls=':',
```
Comment (dbrakenhoff) [inline hydropandas/extensions/plots.py:450]: maybe just use a solid black line? the short and wide dashed lines are difficult to render nicely.

File: hydropandas/extensions/plots.py [hunk 3]
```diff
@@ -307,6 +307,247 @@ def interactive_map(
...
+                            )
+
+            # highlight blind tube on section plot
+            ax_section.plot([plot_x[counter]]*2,
```
Comment (dbrakenhoff) [inline hydropandas/extensions/plots.py:454]: use `zorder` to ensure the markers for statistics and tube characteristics appear above the lines

File: hydropandas/extensions/plots.py [hunk 4]
```diff
@@ -307,6 +307,247 @@ def interactive_map(
...
+
+        # create figure
+        fig = plt.figure(figsize=(15, 5))
+        gs = GridSpec(1, 2, width_ratios=[1, 3],)
+        ax_section = fig.add_subplot(gs[0])
+        ax_obs = fig.add_subplot(gs[1])
+        axes = [ax_section, ax_obs]
+
+        # plot well layout via markers
+        ax_section.scatter(plot_x,
```
Comment (dbrakenhoff) [inline hydropandas/extensions/plots.py:379]: The hollow circle is a bit vague, and marking the tube_top might not be necessary as the top of the line already indicates the position of the tube top.

File: hydropandas/extensions/plots.py [hunk 5]
```diff
@@ -307,6 +307,247 @@ def interactive_map(
...
+            - addtional visual checks:
+                maximum observation is close to or above ground level,
+                maximum observation is close to or above tube top
+                minimum observation is close to or below tube bottom (sand trap)
+            - include some interactive Bokeh fancy
+            - apply an offset when two wells are at same location
+            - limit y-axis of section plot to observations only
+        """
+        
+        import matplotlib.pyplot as plt
```
Comment (dbrakenhoff) [inline hydropandas/extensions/plots.py:361]: These can be moved to the top of the file imo.

File: hydropandas/extensions/plots.py (hunk 1 above)
Comment (HMEUW) [reply hydropandas/extensions/plots.py:547]: Solved in update.

File: hydropandas/extensions/plots.py (hunk 2 above)
Comment (HMEUW) [reply hydropandas/extensions/plots.py:450]: Solved in update.

File: hydropandas/extensions/plots.py (hunk 3 above)
Comment (HMEUW) [reply hydropandas/extensions/plots.py:454]: Solved in update.

File: hydropandas/extensions/plots.py (hunk 4 above)
Comment (HMEUW) [reply hydropandas/extensions/plots.py:379]: Changed to star; and included option to skip markers for well layout.

File: hydropandas/extensions/plots.py (hunk 5 above)
Comment (HMEUW) [reply hydropandas/extensions/plots.py:361]: Solved in update.

File: hydropandas/extensions/plots.py [hunk 6]
```diff
@@ -307,6 +309,251 @@ def interactive_map(
...
+            ax.legend(by_label.values(), by_label.keys())
+
+        fig.suptitle(self._obj.name)
+
+        if fn_save is not None:
+            try:
+                fig.savefig(fn_save, dpi=250, bbox_inches='tight',)
+            except:
+                logger.error(f"Save of figure {name} failed ({fn_save}).")
+
```
Comment (dbrakenhoff) [inline hydropandas/extensions/plots.py:554]: add a `fig.tight_layout()` call here

**Review by dbrakenhoff — CHANGES_REQUESTED** [review]
Nice work! I made a couple of suggestions to improve it further, as you already saw probably. Some were just some thoughts I´d figured I'd share, not necessarily requiring changes, but curious to hear your thoughts.

Two more suggestions here:
- I think we should remove the checking (if obs are near bottom) from this function and move that type of logic to a separate function. If you want to keep the visual warning in this plot I'd prefer a different method than adding a horizontal line (but maybe adding a message to the legend entry?). 
- The other thing is moving the legend outside the plot, e.g. with this code:

```python
        handles, labels = ax_obs.get_legend_handles_labels()
        by_label = dict(zip(labels, handles))
        ax_obs.legend(
            loc=(0, 1),
            frameon=False,
            ncol=6,
            handles=by_label.values(),
            labels=by_label.keys(),
            fontsize="small",
        )
```

File: hydropandas/extensions/plots.py (hunk 4 above)
Comment (dbrakenhoff) [reply hydropandas/extensions/plots.py]: :+1:

File: hydropandas/extensions/plots.py [hunk 7]
```diff
@@ -307,6 +307,247 @@ def interactive_map(
...
+                           label='ground level',
+                           marker='_',
+                           color=section_well_layout_color,
+                           )
+        ax_section.scatter(plot_x,
+                           self._obj.screen_top.values,
+                           section_markersize/2,
+                           label='screen top',
+                           marker='x',
+                           color=section_well_layout_color,
```
[the remark this answers is not in the document]
Comment (HMEUW) [reply hydropandas/extensions/plots.py]: You get a simple plot using keyword argument: plot_well_layout_markers=False.

File: hydropandas/extensions/plots.py (hunk 6 above)
Comment (HMEUW) [reply hydropandas/extensions/plots.py:554]: Added

File: hydropandas/extensions/plots.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (dbrakenhoff) [reply hydropandas/extensions/plots.py]: Yea, perfect!

**Pull request merged**